README.md

```diff
@@ -92,6 +92,9 @@ service serves the purpose of providing the necessary dependencies for `ConfigRe
 methods such as `patch()` and `replace()`.  The service returned by the service manager is bound to
 the file specified in the `config_file` key.
 
+The config that is written is sorted by its keys. To disable this, use`setSortingEnabled` and provide 
+`false` as a parameter.
```

Comment:
There's still a problem here: _where_ does the user call this method, exactly? Particularly as this is primarily an _implementation detail_ for the admin API/UI. In scanning zf-apigility-admin, we generally create `ConfigResource` instances _within factories_ for the various models; as such, this would require providing alternatives for each and every one of these factories. I'm beginning to think we may need a configuration switch that any factory that creates a `ConfigResource` would look at when creating an instance. For BC purposes, this would be disabled by default (to ensure that a minimal set of changes is made to existing configuration files), and users would opt-in to the new behavior.

Thoughts?

Reply:
Ah I see what you mean now. Yes this does sound like a challenge. As far as I understand the changes made here are ok. The question is just how to properly use it in Apigility.

I'm not familiar enough with the architecture behind this but could it be as simple as reducing all the places where an instance of `ConfigResource` is created, to just 1 place by introducing a new class that handles the writing of config resources? Either way this feels like a BC break.

Thinking out of the box. Maybe `setSortingEnabled` should not even exist, or maybe it's not the responsibility of a `ConfigResource` to toggle sorting.